File: Enterprise Compliance Platform/src/types/declaration.ts
export type Screen =
  | "landing"
  | "login"
  | "new-declaration"
  | "my-declarations"
  | "approver-dashboard"
  | "approval-queue"
  | "approval-detail"
  | "declaration-detail"
  | "admin-dashboard"
  | "admin-users"
  | "admin-workflows"
  | "admin-config"
  | "admin-dropdowns"
  | "admin-reports";

export type Role = "teamMember" | "approver" | "admin";

export type StatusType =
  | "Draft"
  | "Pending"
  | "Approved"
  | "Declined"
  | "Escalated"
  | "Info Requested";

export type ApprovalDecision =
  | "return"
  | "accept"
  | "org"
  | "foundation"
  | "decline"
  | null;

export interface User {
  id: string;
  name: string;
  email: string;
  passwordHash: string;
  role: Role;
  teamMemberNumber: string;
  department: string;
  position: string;
  lineManager: string | null;
}

export interface Declaration {
  id: string;
  employee: string;
  employeeId?: string;
  department: string;
  type: string;
  Counterparty: string;
  value: number;
  submitted: string;
  approver: string;
  status: StatusType;
  priority: "High" | "Medium" | "Low";
  description: string;
  relationship: string;
  teamMemberNumber: string;
  lineManager: string;
  position: string;
  receivedGiven: string;
  from: string;
  contactPerson: string;
  biddingProcess: string;
  contractNegotiation?: string;
  occasion: string;
  date: string;
  instances: string;
  publicOfficial: string;
  company?: string;
  team?: string;
  substantiation?: string;
  files?: UploadedFile[];
}

export interface WorkflowStep {
  order: number;
  role: "lineManager" | "hr" | "ceo";
  assignee: string;
  assigneeName: string;
  label: string;
  status: "pending" | "approved" | "declined" | "returned";
  decision: ApprovalDecision;
  notes: string;
  decidedAt: string | null;
}

export interface WorkflowInstance {
  declarationId: string;
  steps: WorkflowStep[];
}

export interface WorkflowRule {
  id: string;
  name: string;
  condition: string;
  priority: number;
  steps: { order: number; role: "lineManager" | "hr" | "ceo"; label: string }[];
}

export interface SystemConfig {
  highValueThreshold: number;
  mediumValueThreshold: number;
  slaEscalationDays: number;
  maxDeclarationsPerCounterparty: number;
  emailTemplate: string;
}

export interface Dropdowns {
  departments: string[];
  categories: string[];
  occasions: string[];
  receivedGiven: string[];
  biddingProcess: string[];
  publicOfficial: string[];
  relationships: string[];
  partyTypes: string[];
}

export interface UploadedFile {
  name: string;
  size: number;
  type: string;
  url: string;
  data?: string;
}

export interface ComplianceTrendPoint {
  month: string;
  approved: number;
  declined: number;
}

export interface TypeBreakdownItem {
  name: string;
  value: number;
  color: string;
}

File: GHEBackend/Models/Declaration.cs
using System.ComponentModel.DataAnnotations;

namespace GHEBackend.Models;

public class Declaration
{
    [Key]
    [MaxLength(30)]
    public string Id { get; set; } = string.Empty;

    [MaxLength(150)]
    public string Employee { get; set; } = string.Empty;

    [MaxLength(30)]
    public string TeamMemberNumber { get; set; } = string.Empty;

    [MaxLength(100)]
    public string LineManager { get; set; } = string.Empty;

    [MaxLength(100)]
    public string Position { get; set; } = string.Empty;

    [MaxLength(100)]
    public string Department { get; set; } = string.Empty;

    [MaxLength(100)]
    public string? Company { get; set; }

    [MaxLength(100)]
    public string? Team { get; set; }

    [MaxLength(50)]
    public string Type { get; set; } = string.Empty;

    [MaxLength(150)]
    public string Counterparty { get; set; } = string.Empty;

    public decimal Value { get; set; }

    [MaxLength(20)]
    public string Submitted { get; set; } = string.Empty;

    [MaxLength(100)]
    public string Approver { get; set; } = string.Empty;

    /// <summary>Draft | Pending | Approved | Declined | Escalated | Info Requested</summary>
    [MaxLength(30)]
    public string Status { get; set; } = string.Empty;

    /// <summary>High | Medium | Low</summary>
    [MaxLength(10)]
    public string Priority { get; set; } = string.Empty;

    [MaxLength(500)]
    public string Description { get; set; } = string.Empty;

    [MaxLength(150)]
    public string Relationship { get; set; } = string.Empty;

    /// <summary>Received | Given</summary>
    [MaxLength(20)]
    public string ReceivedGiven { get; set; } = string.Empty;

    /// <summary>Supplier | Customer | Team Member etc.</summary>
    [MaxLength(100)]
    public string From { get; set; } = string.Empty;

    [MaxLength(100)]
    public string ContactPerson { get; set; } = string.Empty;

    [MaxLength(20)]
    public string BiddingProcess { get; set; } = string.Empty;

    [MaxLength(100)]
    public string? ContractNegotiation { get; set; }

    [MaxLength(100)]
    public string Occasion { get; set; } = string.Empty;

    [MaxLength(20)]
    public string Date { get; set; } = string.Empty;

    [MaxLength(10)]
    public string Instances { get; set; } = string.Empty;

    [MaxLength(10)]
    public string PublicOfficial { get; set; } = string.Empty;

    [MaxLength(500)]
    public string? Substantiation { get; set; }
}
